adapters/handlers/rest/handlers_schema.go

```diff
@@ -250,8 +250,27 @@ func (s *schemaHandlers) restoreSnapshot(params schema.SchemaObjectsSnapshotsRes
 func (s *schemaHandlers) restoreSnapshotStatus(params schema.SchemaObjectsSnapshotsRestoreStatusParams,
 	principal *models.Principal,
 ) middleware.Responder {
-	// TODO implement
-	return nil
+	status, restoreError, path, err := s.manager.RestoreSnapshotStatus(params.HTTPRequest.Context(), principal, params.ClassName, params.StorageName, params.ID)
+	if err != nil {
```

Comment:
I think that here we should handle more error types, like the `Forbidden` errors or the ones that come from `authorizer`. I think that above handler implementations have those cases taken care of, you could look at them to look for an example